cmake/kconfig.cmake

```diff
@@ -104,6 +104,38 @@ if(EXTRA_KCONFIG_OPTIONS)
     )
 endif()
 
+# Set overlay config files list using a list of directories where the
+# overlay files could be found.
+if(DEFINED ENV{OVERLAY_CONFIG_DIRS})
+  set(OVERLAY_CONFIG_DIRS $ENV{OVERLAY_CONFIG_DIRS})
+
+  # We add current directory first so that it is seached first
+  set(overlay_config_dir_list ".:${OVERLAY_CONFIG_DIRS}")
+  string(REPLACE ":" ";" overlay_config_dirs "${overlay_config_dir_list}")
+
+  foreach(f IN ITEMS ${OVERLAY_CONFIG_AS_LIST})
+    if(IS_ABSOLUTE ${f})
+      list(APPEND overlay_config_files ${f})
+    else()
+      set(found 0)
+      foreach(d ${overlay_config_dirs})
+        if(EXISTS ${d}/${f} AND NOT IS_DIRECTORY ${d}/${f})
+          message(STATUS "Overlaying ${d}/${f}")
+          list(APPEND overlay_config_files ${d}/${f})
+          set(found 1)
+          break()
+        endif()
+      endforeach()
+      if(NOT found)
+        # Let the file checker below to report error
+        list(APPEND overlay_config_files ${f})
+      endif()
+    endif()
+  endforeach()
+else()
```

Comment:
Do you need this if-else?

Wouldn't technically . be the highest priority, and therefore the files be appended correctly to overlay_config_files ?

Reply:
Yes, that else() could be removed if needed. Just tried to optimize this a bit so that the default case (no OVERLAY_CONFIG_DIRS defined) would work just like now.

Reply:
Let's get rid of the if-else, this would be consistent with BOARD_ROOT and DTS_BINDINGS_ROOT.